src/main/java/org/embulk/output/mailchimp/MailChimpOutputPluginDelegate.java

```diff
@@ -0,0 +1,144 @@
+package org.embulk.output.mailchimp;
+
+import com.google.common.base.Optional;
+import org.embulk.base.restclient.RestClientOutputPluginDelegate;
+import org.embulk.base.restclient.RestClientOutputTaskBase;
+import org.embulk.base.restclient.jackson.JacksonServiceRequestMapper;
+import org.embulk.base.restclient.jackson.JacksonTopLevelValueLocator;
+import org.embulk.base.restclient.jackson.scope.JacksonAllInObjectScope;
+import org.embulk.base.restclient.record.RecordBuffer;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.config.ConfigDiff;
+import org.embulk.config.ConfigException;
+import org.embulk.config.TaskReport;
+import org.embulk.output.mailchimp.model.AuthMethod;
+import org.embulk.spi.Exec;
+import org.embulk.spi.Schema;
+import org.slf4j.Logger;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.embulk.output.mailchimp.validation.ColumnDataValidator.checkExistColumns;
+
+/**
+ * Created by thangnc on 4/14/17.
+ */
+public class MailChimpOutputPluginDelegate
+        implements RestClientOutputPluginDelegate<MailChimpOutputPluginDelegate.PluginTask>
+{
+    private static final Logger LOG = Exec.getLogger(MailChimpOutputPluginDelegate.class);
+
+    public MailChimpOutputPluginDelegate()
+    {
+    }
+
+    public interface PluginTask
+            extends RestClientOutputTaskBase
+    {
+        @Config("maximum_retries")
+        @ConfigDefault("6")
+        int getMaximumRetries();
+
+        @Config("initial_retry_interval_millis")
+        @ConfigDefault("1000")
+        int getInitialRetryIntervalMillis();
+
+        @Config("maximum_retry_interval_millis")
+        @ConfigDefault("32000")
+        int getMaximumRetryIntervalMillis();
+
+        @Config("timeout_millis")
+        @ConfigDefault("60000")
+        int getTimeoutMillis();
+
+        @Config("auth_method")
+        @ConfigDefault("access_token")
```

Comment:
I am afraid using "access_token" as default "auth_method" will break backwards compatibility. Since "auth_method" was not an option before, any existing config will not have any value. Default to access_token will break existing transfers.